### ISSUE TITLE
StackOverflowError exception when mapping interface

### ISSUE THREAD

**Issue by Sander-Schulenburg** [orika-mapper/orika#188]
After upgrading from 1.4.6 to 1.5.0 I'm getting an StackOverflowError. I have created a simple testcase to reproduce the exception.

**Testcase:**
```
import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;

import java.util.List;

import org.junit.Test;

import ma.glasnost.orika.metadata.Type;
import ma.glasnost.orika.metadata.TypeFactory;

public class SimpleTestCase {

	@Test
	public void testCase() throws Exception {
		Type<?> type = TypeFactory.valueOf(Interface1.class);
		assertThat(type.toString(), is("Interface1<Interface2<Interface2>>"));
	}

	public interface Interface1<T extends Interface2<T>> {
		public List<T> getList();
	}

	public interface Interface2<T extends Interface2<T>> {
		public Interface1<T> getEntity();
	}
}
```

**Exception**
```
java.lang.StackOverflowError
	at ma.glasnost.orika.metadata.TypeFactory.limitedValueOf(TypeFactory.java:334)
	at ma.glasnost.orika.metadata.TypeFactory.limitedValueOf(TypeFactory.java:315)
	at ma.glasnost.orika.metadata.TypeUtil.convertTypeArguments(TypeUtil.java:193)
	at ma.glasnost.orika.metadata.TypeFactory.intern(TypeFactory.java:436)
	at ma.glasnost.orika.metadata.TypeFactory.limitedValueOf(TypeFactory.java:394)
	at ma.glasnost.orika.metadata.TypeFactory.limitedValueOf(TypeFactory.java:377)
	at ma.glasnost.orika.metadata.TypeFactory.limitedValueOf(TypeFactory.java:311)
	at ma.glasnost.orika.metadata.TypeFactory.limitedValueOf(TypeFactory.java:337)
	at ma.glasnost.orika.metadata.TypeFactory.limitedValueOf(TypeFactory.java:315)
	at ma.glasnost.orika.metadata.TypeUtil.convertTypeArguments(TypeUtil.java:193)
	...
	...
	...
	at ma.glasnost.orika.metadata.TypeFactory.intern(TypeFactory.java:436)
	at ma.glasnost.orika.metadata.TypeFactory.limitedValueOf(TypeFactory.java:394)
	at ma.glasnost.orika.metadata.TypeFactory.limitedValueOf(TypeFactory.java:377)
	at ma.glasnost.orika.metadata.TypeFactory.limitedValueOf(TypeFactory.java:311)
	at ma.glasnost.orika.metadata.TypeFactory.limitedValueOf(TypeFactory.java:337)
	at ma.glasnost.orika.metadata.TypeFactory.limitedValueOf(TypeFactory.java:315)
	at ma.glasnost.orika.metadata.TypeUtil.convertTypeArguments(TypeUtil.java:193)
	at ma.glasnost.orika.metadata.TypeFactory.intern(TypeFactory.java:436)
	at ma.glasnost.orika.metadata.TypeFactory.limitedValueOf(TypeFactory.java:394)
	at ma.glasnost.orika.metadata.TypeFactory.limitedValueOf(TypeFactory.java:377)
	at ma.glasnost.orika.metadata.TypeFactory.limitedValueOf(TypeFactory.java:311)
	at ma.glasnost.orika.metadata.TypeFactory.limitedValueOf(TypeFactory.java:337)
	at ma.glasnost.orika.metadata.TypeFactory.limitedValueOf(TypeFactory.java:315)
```

**Comment by papegaaij** [comment]
Please note that his is a not a duplicate of #166. We use a custom build made from master after the commit that fixes #166

**Comment by brabenetz** [comment]
Thanks for the compact TestCase. Helped a lot.
Looks as if this Bug was introduced by fixing another bug for issue #67 in commit @c37c53a (Type evaluation for "Map<Date, Map<Integer, String>>").
The attached Patch File should fix both cases now:
[ISSUE-188_StackOverflowError.patch.txt](https://github.com/orika-mapper/orika/files/736785/ISSUE-188_StackOverflowError.patch.txt)
I also extended the TestCase a little bit:
[Issue188TestCase.java.txt](https://github.com/orika-mapper/orika/files/736784/Issue188TestCase.java.txt)
